From ce9c401c57454b32ea0a45b85ad9796a99466be7 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Tue, 22 May 2018 19:09:20 +0300
Subject: [PATCH 0637/1051] dts: marvell: Enable ap0_spi0 interface on A8K Plus
 boards

Enable AP SPI0 interface on ARMADA 8K+ development boards.
These boards have 16MB flash with resetved 4MB at offset 0x0
for u-boot and reserved last 64KB for the external BootROM image.

Change-Id: Ie4389ed18c7641cda06c8ec80669a7c365ffea2b
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-8080.dtsi  |  1 +
 .../arm64/boot/dts/marvell/armada-8082-db.dts |  5 +++
 .../boot/dts/marvell/armada-8082-db.dtsi      | 31 +++++++++++++++++++
 arch/arm64/boot/dts/marvell/armada-8082.dtsi  |  1 +
 .../arm64/boot/dts/marvell/armada-8084-db.dts |  5 +++
 .../boot/dts/marvell/armada-8084-db.dtsi      | 31 +++++++++++++++++++
 arch/arm64/boot/dts/marvell/armada-8160.dtsi  |  1 +
 .../arm64/boot/dts/marvell/armada-8162-db.dts |  8 +++++
 .../boot/dts/marvell/armada-8162-db.dtsi      | 31 +++++++++++++++++++
 .../arm64/boot/dts/marvell/armada-8164-db.dts |  8 +++++
 .../boot/dts/marvell/armada-8164-db.dtsi      | 31 +++++++++++++++++++
 11 files changed, 153 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-8080.dtsi b/arch/arm64/boot/dts/marvell/armada-8080.dtsi
index 798c772c389d..12411dd85b89 100644
--- a/arch/arm64/boot/dts/marvell/armada-8080.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8080.dtsi
@@ -25,5 +25,6 @@
 	aliases {
 		serial0 = &ap0_uart0;
 		serial1 = &ap0_uart1;
+		spi0 = &ap0_spi0;
 	};
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-8082-db.dts b/arch/arm64/boot/dts/marvell/armada-8082-db.dts
index b0332f1fc121..3ad5a889bd7d 100644
--- a/arch/arm64/boot/dts/marvell/armada-8082-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8082-db.dts
@@ -65,9 +65,14 @@
 		ethernet1 = &cp0_eth1;
 		ethernet2 = &cp0_eth2;
 		ethernet3 = &cp1_eth0;
+		spi0 = &ap0_spi0;
 	};
 };
 
+&ap0_spi0 {
+	status = "okay";
+};
+
 &ap0_xor0 {
 	status = "disabled";
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-8082-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8082-db.dtsi
index 2610977a9ab1..d352e7d5375c 100644
--- a/arch/arm64/boot/dts/marvell/armada-8082-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8082-db.dtsi
@@ -57,6 +57,37 @@
 	status = "okay";
 };
 
+&ap0_spi0 {
+	spi-flash@0 {
+		#address-cells = <0x1>;
+		#size-cells = <0x1>;
+		compatible = "jedec,spi-nor";
+		reg = <0x0>;
+		spi-max-frequency = <20000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x400000>;
+				read-only;
+			};
+			partition@400000 {
+				label = "filesystem";
+				reg = <0x400000 0xbf0000>;
+			};
+			partition@bf0000 {
+				label = "ext-bootrom";
+				reg = <0xff0000 0x10000>;
+				read-only;
+			};
+		};
+	};
+};
+
 &cp0_usb3_0 {
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-8082.dtsi b/arch/arm64/boot/dts/marvell/armada-8082.dtsi
index d2fafb92308e..b72b6913125f 100644
--- a/arch/arm64/boot/dts/marvell/armada-8082.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8082.dtsi
@@ -94,6 +94,7 @@
 	aliases {
 		serial0 = &ap0_uart0;
 		serial1 = &ap0_uart1;
+		spi0 = &ap0_spi0;
 	};
 
 	/* Delete unsupported interfaces from CP nodes:
diff --git a/arch/arm64/boot/dts/marvell/armada-8084-db.dts b/arch/arm64/boot/dts/marvell/armada-8084-db.dts
index 4319cdc45d26..9628bba0fd84 100644
--- a/arch/arm64/boot/dts/marvell/armada-8084-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8084-db.dts
@@ -67,9 +67,14 @@
 		ethernet3 = &cp1_eth0;
 		ethernet4 = &cp2_eth0;
 		ethernet5 = &cp3_eth0;
+		spi0 = &ap0_spi0;
 	};
 };
 
+&ap0_spi0 {
+	status = "okay";
+};
+
 &ap0_xor0 {
 	status = "disabled";
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-8084-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8084-db.dtsi
index 9c8da929363d..8872eb0c58ce 100644
--- a/arch/arm64/boot/dts/marvell/armada-8084-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8084-db.dtsi
@@ -57,6 +57,37 @@
 	status = "okay";
 };
 
+&ap0_spi0 {
+	spi-flash@0 {
+		#address-cells = <0x1>;
+		#size-cells = <0x1>;
+		compatible = "jedec,spi-nor";
+		reg = <0x0>;
+		spi-max-frequency = <20000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x400000>;
+				read-only;
+			};
+			partition@400000 {
+				label = "filesystem";
+				reg = <0x400000 0xbf0000>;
+			};
+			partition@bf0000 {
+				label = "ext-bootrom";
+				reg = <0xff0000 0x10000>;
+				read-only;
+			};
+		};
+	};
+};
+
 &cp0_usb3_0 {
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-8160.dtsi b/arch/arm64/boot/dts/marvell/armada-8160.dtsi
index 64d6cf0fe879..f328d7bd32e2 100644
--- a/arch/arm64/boot/dts/marvell/armada-8160.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8160.dtsi
@@ -78,5 +78,6 @@
 	aliases {
 		serial0 = &ap0_uart0;
 		serial1 = &ap0_uart1;
+		spi0 = &ap0_spi0;
 	};
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-8162-db.dts b/arch/arm64/boot/dts/marvell/armada-8162-db.dts
index 3728f03ac0a9..8d956ac5e28e 100644
--- a/arch/arm64/boot/dts/marvell/armada-8162-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8162-db.dts
@@ -59,6 +59,14 @@
 		device_type = "memory";
 		reg = <0x0 0x0 0x0 0x80000000>;
 	};
+
+	aliases {
+		spi0 = &ap0_spi0;
+	};
+};
+
+&ap0_spi0 {
+	status = "okay";
 };
 
 &ap0_xor0 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8162-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8162-db.dtsi
index 6e6e72be75bb..faaa3e061c17 100644
--- a/arch/arm64/boot/dts/marvell/armada-8162-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8162-db.dtsi
@@ -57,6 +57,37 @@
 	status = "okay";
 };
 
+&ap0_spi0 {
+	spi-flash@0 {
+		#address-cells = <0x1>;
+		#size-cells = <0x1>;
+		compatible = "jedec,spi-nor";
+		reg = <0x0>;
+		spi-max-frequency = <20000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x400000>;
+				read-only;
+			};
+			partition@400000 {
+				label = "filesystem";
+				reg = <0x400000 0xbf0000>;
+			};
+			partition@bf0000 {
+				label = "ext-bootrom";
+				reg = <0xff0000 0x10000>;
+				read-only;
+			};
+		};
+	};
+};
+
 &cp0_usb3_0 {
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-8164-db.dts b/arch/arm64/boot/dts/marvell/armada-8164-db.dts
index ee97bd8533db..6aa824c34e67 100644
--- a/arch/arm64/boot/dts/marvell/armada-8164-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8164-db.dts
@@ -59,6 +59,14 @@
 		device_type = "memory";
 		reg = <0x0 0x0 0x0 0x80000000>;
 	};
+
+	aliases {
+		spi0 = &ap0_spi0;
+	};
+};
+
+&ap0_spi0 {
+	status = "okay";
 };
 
 &ap0_xor0 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8164-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8164-db.dtsi
index 9a0f75fa2d8b..76cd41a1e52d 100644
--- a/arch/arm64/boot/dts/marvell/armada-8164-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8164-db.dtsi
@@ -57,6 +57,37 @@
 	status = "okay";
 };
 
+&ap0_spi0 {
+	spi-flash@0 {
+		#address-cells = <0x1>;
+		#size-cells = <0x1>;
+		compatible = "jedec,spi-nor";
+		reg = <0x0>;
+		spi-max-frequency = <20000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x400000>;
+				read-only;
+			};
+			partition@400000 {
+				label = "filesystem";
+				reg = <0x400000 0xbf0000>;
+			};
+			partition@bf0000 {
+				label = "ext-bootrom";
+				reg = <0xff0000 0x10000>;
+				read-only;
+			};
+		};
+	};
+};
+
 &cp0_usb3_0 {
 	status = "okay";
 };
-- 
2.17.1

